### ISSUE TITLE
Strange Error talking to server

### ISSUE THREAD

**Issue by zhanhuaih** [OmniSharp/omnisharp-sublime#192]
I am facing a strange error here. Omnisharp works perfectly when the computer first starts. If then I quit sublime text for like 10 minutes, as I reopen my c# project Omnisharp will start misbehaving and keep having trouble talking to the Omnisharp Server. So no functionality provided. Just don't understand why this could happen.

Asking for server output gives me nothing (nothing comes up) once Omnisharp enters this strange state.
![2015-07-01 11 46 32](https://cloud.githubusercontent.com/assets/134088/8455929/6e954aa0-204b-11e5-8d06-df0bca537a11.png)
(It also fails talking to /checkcode etc)

**Comment by HalfVoxel** [comment]
I found that if I close sublime text and quickly reopen it (within a few seconds). It will completely fail. Likely because the port is not freed yet (just speculation on my part, but it seems reasonable). I seems a bit weird that an os would lock the port for a whole 10 minutes but that's the only explanation I can think of.
Can you check what the server output is?

**Comment by jchannon** [comment]
A bit rubbish but try killing off mono process and sublime and reopen
sublime. Sorry

**Comment by zhanhuaih** [comment]
Ah, that worked.
What is the actual problem though?

**Comment by fishg** [comment]
Because socket port conflict with old mono background process killed delay?

**Comment by zhanhuaih** [comment]
Ah, no! The success last time was a miracle. I just closed sublime text and restarted it, and OmniSharp is getting into trouble again.

This time I cannot find any mono processes (I am using ps aux|grep mono). I killed sublime and restarted but it keeps failing.

**Comment by zhanhuaih** [comment]
I even tried removing and reinstalling the mono package. No luck. After letting my mac slept for like 10hours, Omnisharp still fails in the same way as I open sublime.

Omnisharp is such a great tool and I don't want to switch back to visual studio so any help will be greatly appreciated!!

**Comment by fishg** [comment]
@zhanhuaih 
可以中文？启动sublime，打开你的c#相关工程后，等10秒左右，把如下图所示的build log贴出来看看
![qq20150702-2](https://cloud.githubusercontent.com/assets/1423545/8469710/118d6fc4-20b2-11e5-91c6-4b68da59de5c.png)

**Comment by zhanhuaih** [comment]
![2015-07-02 2 04 39](https://cloud.githubusercontent.com/assets/134088/8469746/54129796-20c3-11e5-8d2b-892b7742ddb1.png)
@fishg 谢谢，但是...

**Comment by zhanhuaih** [comment]
It is weird but I found I solution somehow. All I needed is just to do what https://github.com/OmniSharp/omnisharp-sublime#project-setting says. Then just start sublime (**before you close the sublime app itself you need to close all windows containing C# files or as sublime reopens any window with a opened C# file will sometimes trigger Omnisharp to run and then it will fail**), open the sublime project file and Omnisharp will be ok.

My project only contains 1 project.json so I suggest there are problems with 
> The server will automatically find the the solution file from the folder you have opened in Sublime.

Hope this helps.

**Comment by jchannon** [comment]
I'm not sure how well works with DNX apps. If I were you I'd use Atom or
feel free to continue the effort to get Sublime and DNX working :)

**Comment by L4D15** [comment]
I'm using this package to work with Unity and I'm getting the same error when loading the project. Any advice?

I installed mono through homebrew, if that matters.

**Comment by L4D15** [comment]
Ok, I figured it out. I removed and installed Mono using brew, and linked it. Maybe I was missing the "link" step the last time I installed it.

**Comment by robbert229** [comment]
This error comes from being unable to start the server. This is usually caused by a bad mono install. 

If you have any more problems don't hesitate to open up an issue :)

**Comment by mandarinx** [comment]
Sorry for resurrecting this issue, but I wanted to add that I had the same error and solved it by cleaning up the config.json file. I had a dangling comma in the IgnoredCodeIssues array which freaked out OmniSharp.

**Comment by dmalpica** [comment]
To clarify, attempting to use Mono from the .pkg on the Mono website was causing this error for me. 

ChaimHong is correct, one way to do it is change the path looking for Mono in the omnisharp script inside the omnisharp sublime package at:

/Users/YOU/Library/Application Support/Sublime Text 3/Packages/OmniSharp/PrebuiltOmniSharpServer/omnisharp

The cleaner alternative for me was to install via Homebrew. 

To uninstall the .pkg, I followed the steps here: http://www.mono-project.com/docs/about-mono/supported-platforms/osx/#uninstalling-mono-on-mac-os-x

With mono installed via Homebrew the path issue was resolved. See normal build results and status line:

![screen shot 2017-02-15 at 3 35 02 pm](https://cloud.githubusercontent.com/assets/6834522/23000219/5d4ad0c6-f394-11e6-830d-fbe4a60542b0.png)
